assets/src/components/app.tsx

```diff
@@ -20,28 +19,37 @@ import Modal from "./modal"
 import SettingsPage from "./settingsPage"
 import ShuttleMapPage from "./shuttleMapPage"
 import LateView from "./lateView"
-import { OpenView } from "../state"
 import { allOpenRouteIds } from "../models/routeTab"
 import Nav from "./nav"
 import RightPanel from "./rightPanel"
 import { mapModeForUser } from "../util/mapMode"
 import { Ghost, VehicleInScheduledService } from "../realtime"
 import MapPage from "./mapPage"
 import SearchPage from "./searchPage"
+import { OpenView, isPagePath } from "../state/pagePanelState"
+import { usePanelStateFromStateDispatchContext } from "../hooks/usePanelState"
 
 export const AppRoutes = () => {
   useAppcues()
+  const location = useLocation()
 
-  const [{ openView, routeTabs, selectedVehicleOrGhost }] =
-    useContext(StateDispatchContext)
+  const [{ routeTabs }] = useContext(StateDispatchContext)
 
-  const { socket }: { socket: Socket | undefined } = useContext(SocketContext)
+  const {
+    setPath,
+    currentView: { openView, selectedVehicleOrGhost },
+  } = usePanelStateFromStateDispatchContext()
 
-  const location = useLocation()
+  // Keep panel in sync with current path
+  const { pathname: path } = location
+  useEffect(() => {
+    isPagePath(path) && setPath(path)
+  }, [path, setPath])
```

Comment:
**question:** Is there an explicit test for this logic anywhere?

Reply:
good question, there is not yet, as I wasn't sure how to test stuff with the browser router and such. I'll look into it 👍🏻

Reply:
I have a test that asserts this works, but I don't have a test that asserts that this doesn't do anything on pages like `/settings`.

I'll go ahead and commit the affirmative test, but I'm not sure if we should keep the one that doesn't work but skip it for now?

## Background
So I've found `app.test.tsx` to be somewhat fragile, I'm honestly still not got my head wrapped around all of this.

My main issue has been that depending on which tests are run, and in which order, some state is being saved across tests within the same module.
I _think_ this is due to JSDOM, because JSDOM is only ["reconfigured"](https://github.com/jestjs/jest/issues/5987) when the test module changes, but not in-between tests.
This means, that if you add a test before the `"render"` test, that "navigates" somewhere using the browser API, that `window` state will not be reset until the next module, and the snapshot changes because the `window.location` is different (and I'm still unclear on how `MemoryRouter` interacts with this). If `window.location` is different, that seems to affect the router, causing snapshot tests to fail because the dom would be in a different state due to being on a different page.

Some of the suggestions in the linked jest issue, suggest using a `reconfigure` method on JSDOM, but I can't seem to find any way to get access to the created `DOM` object that `jest-environment-jsdom` creates https://github.com/jestjs/jest/blob/main/packages/jest-environment-jsdom/src/index.ts#L34 .

I don't think any of the existing tests in `app.test.tsx` actually test navigation within the application, like this new test is. I think this means that since the new test tests navigation, if we were to randomize the order of the tests in this module (something I'd like to turn on in jest), we'd start having random tests fail due to this.

Reply:
10f03586dc03b14cd18fbf69fc58b44ea4010f43

Reply:
btw this is what the other test (that fails) looks like
```tsx
  test.skip("doesn't update panel state when page is not an allowed page", async () => {
    const mockedUsePanelState = mockUsePanelState()
    render(
      <StateDispatchProvider state={stateFactory.build()} dispatch={jest.fn()}>
        <MemoryRouter initialEntries={["/"]}>
          <AppRoutes />
        </MemoryRouter>
      </StateDispatchProvider>
    )

    await userEvent.click(screen.getByRole("link", { name: "Settings" }))

    expect(mockedUsePanelState().setPath).toHaveBeenNthCalledWith(1, "/")
    expect(mockedUsePanelState().setPath).toHaveBeenCalledTimes(1)
  })
  ```